analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -76,105 +69,99 @@ protected override void Initialize(SonarAnalysisContext context)
         {
-            if (sideA.ToStringContains("GetType") &&
-                sideB is TypeOfExpressionSyntax sideBeTypeOf &&
-                sideBeTypeOf.Type is { } typeSyntax &&
-                (sideA as InvocationExpressionSyntax).IsGetTypeCall(context.SemanticModel) &&
-                context.SemanticModel.GetTypeInfo(typeSyntax).Type is { } typeSymbol &&
-                typeSymbol.IsSealed &&
-                !typeSymbol.OriginalDefinition.Is(KnownType.System_Nullable_T))
+            if (sideA.ToStringContains("GetType")
+                && sideB is TypeOfExpressionSyntax sideBeTypeOf
+                && (sideA as InvocationExpressionSyntax).IsGetTypeCall(context.SemanticModel)
+                && context.SemanticModel.GetTypeInfo(sideBeTypeOf.Type).Type is { } typeSymbol // Can be null for empty identifier from 'typeof' unfinished syntax
+                && typeSymbol.IsSealed
+                && !typeSymbol.OriginalDefinition.Is(KnownType.System_Nullable_T))
             {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location, MessageIsOperator));
+                ReportDiagnostic(context, MessageIsOperator);
             }
         }
 
-        private static void CheckForIsInstanceOfType(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax invocation,
-            MemberAccessExpressionSyntax memberAccess, IMethodSymbol methodSymbol)
+        private static void CheckForIsInstanceOfType(SyntaxNodeAnalysisContext context, MemberAccessExpressionSyntax memberAccess, IMethodSymbol methodSymbol)
         {
-            if (methodSymbol.Name != "IsInstanceOfType")
+            if (methodSymbol.Name == "IsInstanceOfType" && memberAccess.Expression is TypeOfExpressionSyntax)
             {
-                return;
+                ReportDiagnostic(context, MessageIsOperator, true);
             }
+        }
 
-            if (memberAccess.Expression is TypeOfExpressionSyntax)
+        private static void CheckForIsAssignableFrom(SyntaxNodeAnalysisContext context, MemberAccessExpressionSyntax memberAccess, IMethodSymbol methodSymbol, ExpressionSyntax argument)
+        {
+            if (methodSymbol.Name == nameof(Type.IsAssignableFrom) && (argument as InvocationExpressionSyntax).IsGetTypeCall(context.SemanticModel))
             {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, invocation.GetLocation(),
-                    ImmutableDictionary<string, string>.Empty
-                        .Add(UseIsOperatorKey, true.ToString())
-                        .Add(ShouldRemoveGetType, false.ToString()),
-                    MessageIsOperator));
+                if (memberAccess.Expression is TypeOfExpressionSyntax)
+                {
+                    ReportDiagnostic(context, MessageIsOperator, true, true);
+                }
+                else
+                {
+                    ReportDiagnostic(context, MessageIsInstanceOfType, false, true);
```

Comment:
note to self: I'll need to revisit this refactoring again

Reply:
I think named arguments may help with readability here and when the arguments are passed explicitly